numpy/lib/npyio.py

```diff
@@ -1106,18 +1106,20 @@ def savetxt(fname, X, fmt='%.18e', delimiter=' ', newline='\n', header='',
 
         # Handle 1-dimensional arrays
         if X.ndim == 1:
-            # Common case -- 1d array of numbers
-            if X.dtype.names is None:
-                X = np.atleast_2d(X).T
-                ncol = 1
+            X = np.atleast_2d(X).T
 
-            # Complex dtype -- each field indicates a separate column
-            else:
-                ncol = len(X.dtype.descr)
-        else:
-            ncol = X.shape[1]
+        nrow = X.shape[0]
```

Comment:
Ok, using len(X.dtype.names), would fix this, right?
--
Sent from myMail app for Android